cblite-tests/e2e/listener-test.ts

```diff
@@ -12,58 +12,1758 @@ export class ListenerTests extends TestCase {
   }
 
   /**
-   *
-   * @returns {Promise<ITestResult>} A promise that resolves to an ITestResult object which contains the result of the verification.
+   * Test Collection Change Listener with OLD API (collection.removeChangeListener)
+   * 
+   * This test verifies backward compatibility - the old way of removing listeners
+   * should still work even though we now return ListenerToken objects.
+   * 
+   * @returns {Promise<ITestResult>} A promise that resolves to an ITestResult object
    */
-  async testCollectionChange(): Promise<ITestResult> {
+  async testCollectionChangeListenerOldAPI(): Promise<ITestResult> {
     try {
-      const results: string[] = [];
-      const docId1 = "doc1";
-      const docId2 = "doc2";
       const collection = await this.database.createCollection(
-        "testCollectionChange",
-        "testScope"
+        'testCollOldAPI',
+        'testScope'
       );
 
-      // Create a promise that resolves when the listener is added
+      let changeCount = 0;
+      const documentIds: string[] = [];
+
+      // Add listener - returns ListenerToken
       const token = await collection.addChangeListener((change) => {
-        for (const doc of change.documentIDs) {
-          results.push(doc);
-        }
+        changeCount++;
+        documentIds.push(...change.documentIDs);
       });
 
-      //create documents to trigger the change listener
+      // Verify token is ListenerToken object (not string anymore)
+      expect(token).to.not.be.a('string');
+      expect(token.getUuidToken).to.be.a('function');
```

Comment:
First, I don't know much about TS. Is this method also available publicly as a public API?